html/semantics/forms/form-submission-0/form-double-submit-to-different-origin-frame.html WPT test fails to run with our infrastructure

cdumez opened the pull request:
#### d883156c3539a453b1f2f903d0da1bfda5ac306a
<pre>
html/semantics/forms/form-submission-0/form-double-submit-to-different-origin-frame.html WPT test fails to run with our infrastructure
<a href="https://bugs.webkit.org/show_bug.cgi?id=243843">https://bugs.webkit.org/show_bug.cgi?id=243843</a>

Reviewed by Tim Nguyen.

The test relies on switching a URL cross-origin by switching its scheme from http to https. However,
when running WPT tests on an HTTP server that uses non-default ports, this doesn&apos;t result in a valid
URL since the port wasn&apos;t updated to the HTTPS port.

To address the issue, rely on get_host_info().REMOTE_ORIGIN to get a remote origin URL.

Upstream PR: <a href="https://github.com/web-platform-tests/wpt/pull/35441">https://github.com/web-platform-tests/wpt/pull/35441</a>

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-to-different-origin-frame-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-to-different-origin-frame.html:

Canonical link: <a href="https://commits.webkit.org/253357@main">https://commits.webkit.org/253357@main</a>
</pre>

nt1m reviewed (approved):
Looks like you also fixed the test for Firefox: https://github.com/web-platform-tests/wpt/pull/35441/checks?check_run_id=7795091352

cdumez committed:
html/semantics/forms/form-submission-0/form-double-submit-to-different-origin-frame.html WPT test fails to run with our infrastructure
https://bugs.webkit.org/show_bug.cgi?id=243843

Reviewed by Tim Nguyen.

The test relies on switching a URL cross-origin by switching its scheme from http to https. However,
when running WPT tests on an HTTP server that uses non-default ports, this doesn't result in a valid
URL since the port wasn't updated to the HTTPS port.

To address the issue, rely on get_host_info().REMOTE_ORIGIN to get a remote origin URL.

Upstream PR: https://github.com/web-platform-tests/wpt/pull/35441

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-to-different-origin-frame-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/semantics/forms/form-submission-0/form-double-submit-to-different-origin-frame.html:

Canonical link: https://commits.webkit.org/253357@main
